cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -97,6 +103,29 @@ func NewServer(configGetter server.KubernetesConfigGetter) *Server {
 			}
 			return typedClient, dynamicClient, nil
 		},
+		actionConfigGetter: func(ctx context.Context, namespace string) (*action.Configuration, error) {
+			if configGetter == nil {
+				return nil, status.Errorf(codes.Internal, "configGetter arg required")
+			}
+			config, err := configGetter(ctx)
+			if err != nil {
+				return nil, status.Errorf(codes.FailedPrecondition, fmt.Sprintf("unable to get config : %v", err))
+			}
+
+			restClientGetter := agent.NewConfigFlagsFromCluster(namespace, config)
+			clientSet, err := kubernetes.NewForConfig(config)
```

Comment:
Just a thought: are we creating yet another config here? Shouldn't we use the `typedClient` which already is doing ` `kubernetes.NewForConfig(config)` I mean, atm I don't know if it returns the same instance or it creates an additional one.

Reply:
There is no additional client created here because `GetInstalledPackageDetails` doesn't even use `clientGetter` (since it doesn't require a k8s client outside of the helm client). But yes, if a handler did need both, the helm action config would have a separate client (in memory) from the typed client used directly by the handler. But I don't see an issue with this?

We could refactor things so that the `actionConfigGetter` is passed a typed client, but then it'd be additionally creating the dynamic client unnecessarily. I think the `clientGetter` is only returning a tuple of clients because of the history here (it was the interface originally between the plugins and the core), so we could update to just provide separate getters `typedClientGetter` and `dynamicClientGetter` that can be used by our functions.

Reply:
I agree, I don't see any issue with it, given that both use the same config so, even they are different objects in memory, they'll be the same thing in practice.
Mmm, yep, perhaps adding separate getters could be a good refactor option.
Anyway, it just was a thought, nothing important or critical.